Sources/TootSDK/TootClient/TootClient+Reports.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  TootClient+Reports.swift
+//  
+//
+//  Created by Łukasz Rutkowski on 26/10/2023.
+//
+
+import Foundation
+
+public extension TootClient {
+
+    /// Report problematic users or posts to moderators.
+    ///
+    /// - Warning: For Pixelfed use `PixelfedReportParams`.
+    @discardableResult
+    func report(_ params: ReportParams) async throws -> Report {
```

Comment:
I think we need to find a way to use the same signature for requesting the report from all flavours. Change in behaviour based on flavour should be transparent for the client. We have examples of such methods `followAccountURI` and `updateCredentials`.

In this case, the params object could be a higher-level API provided by TootSDK which (depending on flavour) we will use internally to create the appropriate server request.